server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -290,6 +290,18 @@ private void parseGeoPointIgnoringMalformed(ParseContext context, GeoPoint spars
         }
     }
 
+    @Override
+    public List<?> parseSourceValue(Object value) {
+        if (value instanceof List) {
+            try {
+                GeoUtils.parseGeoPoint(value, ignoreZValue.value());
+            } catch (ElasticsearchParseException e) {
+                return (List<?>) value;
```

Comment:
Wow! I get why we're doing this but wow!

Reply:
Yes, this was not a proud moment 😊 Maybe I can instead introduce a method like `GeoUtils.isSingleGeoPoint`. I'll need general guidance from the geo experts on the best way to return geo data, I'll make sure to loop them in on the follow-up PR.